### ISSUE TITLE
Predicate created by `anyPass` can't be used with array `.filter`

### ISSUE THREAD

**Issue by rakeshpai** [selfrefactor/rambda#642]
The following example gives a compile error:

```ts
import { anyPass, filter } from "rambda";

const isGt5 = (num: number) => num > 5;

const pred = anyPass([isGt5]);

const xs = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];

const filtered1 = filter(pred)(xs); // works
const filtered2 = xs.filter(pred); // compile error
```

This is the compile error: 

```
No overload matches this call.
  Overload 1 of 2, '(predicate: (value: number, index: number, array: number[]) => value is number, thisArg?: any): number[]', gave the following error.
    Argument of type 'SafePred<number>' is not assignable to parameter of type '(value: number, index: number, array: number[]) => value is number'.
```

You can see this example in action here: https://codesandbox.io/s/pedantic-jennings-ret7c2?file=/src/index.ts

I think this is because of the type definition of [`SafePred`](https://github.com/selfrefactor/rambda/blob/d2faada92bf1c10c7287d2567e974bd3e96cfd65/index.d.ts#L33), which is incompatible with the definition of the function taken in by array `.filter`. I'm not sure why `SafePred` is defined this way, with `(...x: T[])` as the arguments. Shouldn't it be just `(x: T)`?

**Comment by selfrefactor** [comment]
I will into it. Thank you for the detailed report; that makes things easier.

**Comment by selfrefactor** [comment]
It is fixed and will be released with `7.2.0`

**Comment by rakeshpai** [comment]
Thanks a ton, @selfrefactor!

**Comment by selfrefactor** [comment]
Closing the issue as fix was released with `7.2.0`
